CMSgov/easi-app
# K1m 581 see request notes

# ES-581

## Changes proposed in this pull request

Adds the existing notes list in the "Notes" tab for the 508 user/tester.  It includes a screen-reader-only link to navigate to the notes region.

## Reviewer Notes

~From what I can tell, if the user has only an EUA ID (and does not have a name), the fn to fetch the user info applies the EUA as lowercase~ I change the default CommonName to have a last name so that it was more apparent that it was being rendered.

I added `fetchPolicy: network-only` to the accessibility request query because the page was re-rendering, which meant that after adding a note, the page was going to the default Documents tab instead of staying on the notes. ([see Slack discussion](https://trussworks.slack.com/archives/C01BCRE1Q20/p1623970736320900))

I created a new file for the react testing library tests (`AccessibilityRequestDetailPageRTL.test.tsx`).  In a future PR or story we can rewrite the rest of the tests for that page and then rename the file to the original name (`AccessibilityRequestDetailPageRTL.test.tsx`).

## Setup

- Log in as a 508 user/tester
- Go to an accessibility request
- Click on the notes tab and add some notes
- Log out and log in as a different user and go to the same request
- Check that the author of the note corresponds to the corresponding user's name

- Also log in as a business owner.  Ensure that "notes" (in Apollo ext) are not included in the accessibility request query returned data.

## Code Review Verification Steps

### As the original developer, I have

- [x] Tested user-facing changes with voice-over
- [x] Checked for landmarks, page heading structure and links using [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu)

### As the code reviewer, I have


### As a designer, I have

- [x] Checked accessibility against criteria

## Screenshots

![image](https://user-images.githubusercontent.com/13249580/123116673-f5401d00-d3f5-11eb-932b-a351f230ccc5.png)
